File: ts-reference/examples/basic-2/generated-client/instructions/create.ts
import { TransactionInstruction, PublicKey } from "@solana/web3.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import BN from "bn.js" // eslint-disable-line @typescript-eslint/no-unused-vars
import * as borsh from "@project-serum/borsh" // eslint-disable-line @typescript-eslint/no-unused-vars
import { PROGRAM_ID } from "../programId"

export interface CreateArgs {
  authority: PublicKey
}

export interface CreateAccounts {
  counter: PublicKey
  user: PublicKey
  systemProgram: PublicKey
}

export const layout = borsh.struct([borsh.publicKey("authority")])

export function create(args: CreateArgs, accounts: CreateAccounts) {
  const keys = [
    { pubkey: accounts.counter, isSigner: true, isWritable: true },
    { pubkey: accounts.user, isSigner: true, isWritable: true },
    { pubkey: accounts.systemProgram, isSigner: false, isWritable: false },
  ]
  const identifier = Buffer.from([24, 30, 200, 40, 5, 28, 7, 119])
  const buffer = Buffer.alloc(1000)
  const len = layout.encode(
    {
      authority: args.authority,
    },
    buffer
  )
  const data = Buffer.concat([identifier, buffer]).slice(0, 8 + len)
  const ix = new TransactionInstruction({ keys, programId: PROGRAM_ID, data })
  return ix
}

File: examples/client-gen/basic_2/instructions/create.py
from __future__ import annotations
import typing
from solders.pubkey import Pubkey
from solders.sysvar import RENT
from solders.instruction import Instruction, AccountMeta
from anchorpy.borsh_extension import BorshPubkey
import borsh_construct as borsh
from ..program_id import PROGRAM_ID


class CreateArgs(typing.TypedDict):
    authority: Pubkey


layout = borsh.CStruct("authority" / BorshPubkey)


class CreateAccounts(typing.TypedDict):
    counter: Pubkey


def create(
    args: CreateArgs,
    accounts: CreateAccounts,
    program_id: Pubkey = PROGRAM_ID,
    remaining_accounts: typing.Optional[typing.List[AccountMeta]] = None,
) -> Instruction:
    keys: list[AccountMeta] = [
        AccountMeta(pubkey=accounts["counter"], is_signer=False, is_writable=True),
        AccountMeta(pubkey=RENT, is_signer=False, is_writable=False),
    ]
    if remaining_accounts is not None:
        keys += remaining_accounts
    identifier = b"\x18\x1e\xc8(\x05\x1c\x07w"
    encoded_args = layout.build(
        {
            "authority": args["authority"],
        }
    )
    data = identifier + encoded_args
    return Instruction(program_id, data, keys)
